### PR TITLE
BC-2599 - editor component

### PR DESCRIPTION
**Pull request by christian-darsow**
# Short Description

Implement an editor component for task creation 

## Links to Ticket and related Pull-Requests
https://ticketsystem.dbildungscloud.de/browse/BC-2599

As a pre-requirement we needed to setup a repo for our custom CKEditor
https://github.com/hpi-schul-cloud/ckeditor

The builds are published to
https://www.npmjs.com/package/@hpi-schul-cloud/ckeditor

## Checklist before merging

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

File: src/components/molecules/Editor.vue [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+			config,
+			handleInput,
+		};
+	},
+});
+</script>
+
+<style scoped>
+.ck-wrapper {
+	margin: var(--space-xs) 0 var(--space-md) 0;
```
Comment (odalys-dataport) [inline src/components/molecules/Editor.vue:123]: Please use vuetify classes instead

File: src/components/molecules/Editor.vue [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+			}
+
+			return i18n.locale;
+		})();
+
+		const config = {
+			toolbar: {
+				items: (() => {
+					if (props.mode === "simple") {
+						return ["bold", "italic", "underline"];
```
Comment (odalys-dataport) [inline src/components/molecules/Editor.vue:64]: I'm curious about the decision of what to put into the simple mode and why

**Review by odalys-dataport — CHANGES_REQUESTED** [review]

File: src/components/molecules/Editor.vue [hunk 3]
```diff
@@ -0,0 +1,125 @@
+<template>
+	<div>
+		<label class="v-label theme--light">{{ label }}</label>
+
```
[the remark this answers is not in the document]
Comment (christian-darsow) [reply src/components/molecules/Editor.vue:4]: I totally agree :)

File: src/components/molecules/Editor.vue [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+			validator: (value) => ["simple", "regular"].includes(value),
+			default: "regular",
+		},
+	},
+	setup(props, { emit }) {
+		const i18n = inject("i18n");
+
+		const content = ref(props.value);
+		const language = (() => {
+			// as we don't have a translation for ua yet we map to en
```
[the remark this answers is not in the document]
Comment (christian-darsow) [reply src/components/molecules/Editor.vue:52]: There is no such decision. It's just that the translations for en + de + es are already done by CK itself - and for UA we will need to do this on our own in a follow-up ticket. maybe we can even give that back to the community?

File: src/components/molecules/Editor.vue (hunk 2 above)
Comment (christian-darsow) [reply src/components/molecules/Editor.vue:64]: This decision came directly out of my head :) 
But yes, simply there is no decision about modes and features in modes so far
So we might also end up with more then two modes

File: src/components/molecules/Editor.vue (hunk 1 above)
Comment (christian-darsow) [reply src/components/molecules/Editor.vue]: I removed the styling for now until we have an UI

File: src/components/molecules/Editor.vue [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+
+<script>
+import { defineComponent, ref, inject } from "@vue/composition-api";
+import CKEditor from "@ckeditor/ckeditor5-vue2";
+require("@hpi-schul-cloud/ckeditor/build/translations/en");
+require("@hpi-schul-cloud/ckeditor/build/translations/es");
+import CustomCKEditor from "@hpi-schul-cloud/ckeditor";
+
+// eslint-disable-next-line vue/require-direct-export
+export default defineComponent({
```
[the remark this answers is not in the document]
Comment (christian-darsow) [reply src/components/molecules/Editor.vue:23]: I'm not sure about that. For vue3 it seems also to be fine for JS components https://vuejs.org/guide/typescript/overview.html#definecomponent -> looking at the tipp there. If the same goes for the plugin we use in vue2 stays unclear to me. Lets talk again on that :)

**Review by odalys-dataport — APPROVED** [review]

**Pull request merged**